### PR TITLE
Don't use recursive Make (2)

### LINKED ISSUE #2969: eliminate recursive makes as much as possible
See [Recursive Make Considered Harmful](https://accu.org/journals/overload/14/71/miller_2004/), Peter Miller, Overload, 14(71):20-30, February 2006

I've started to eliminate some of the uses of recursive make, which TSK  uses much more than it should.

### PR DESCRIPTION
**Pull request by uckelman-sf**
This removes the last recursive use of Make. Closes #2969.

### CODE REVIEW

**Review by simsong — APPROVED** [review]
whoops! Sorry if that was my fault.

**Review by simsong — APPROVED** [review]

**Review by simsong — APPROVED** [review]
okay, but how are we going to assure that the disk images have been downloaded?

**Comment by simsong** [comment]
@sleuthkit/autopsy-ro - don't like in fiwalk_main.o:
```
d: warning: -bind_at_load is deprecated on macOS
ld: warning: ignoring duplicate libraries: '-lc++'
duplicate symbol '_main' in:
    /Users/runner/work/sleuthkit/sleuthkit/unit_tests/runner-runner.o
    /Users/runner/work/sleuthkit/sleuthkit/tools/fiwalk/src/unit_tests_runner-fiwalk_main.o
ld: 1 duplicate symbols
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by uckelman-sf** [comment]
> okay, but how are we going to assure that the disk images have been downloaded?

Is this on the right issue? There aren't any disk images or downloading here.

**Comment by simsong** [comment]
> > okay, but how are we going to assure that the disk images have been downloaded?
> 
> Is this on the right issue? There aren't any disk images or downloading here.

The `fiwalk_test.cpp` that is now part of the unit test requires disk images.

**Comment by uckelman-sf** [comment]
> > > okay, but how are we going to assure that the disk images have been downloaded?
> > 
> > 
> > Is this on the right issue? There aren't any disk images or downloading here.
> 
> The `fiwalk_test.cpp` that is now part of the unit test requires disk images.

Oh. I had not realized that `runner` was not yet being run as part of `make check`. I'll back that part out here and fix that in another PR.

**Comment by simsong** [comment]
> > > > okay, but how are we going to assure that the disk images have been downloaded?
> > > 
> > > 
> > > Is this on the right issue? There aren't any disk images or downloading here.
> > 
> > 
> > The `fiwalk_test.cpp` that is now part of the unit test requires disk images.
> 
> Oh. I had not realized that `runner` was not yet being run as part of `make check`. I'll back that part out here and fix that in another PR.

Thanks. I will likely have `runner` run the to-be-written python program that downloads the disk images if they do not exist.

**Comment by uckelman-sf** [comment]
> > > > > okay, but how are we going to assure that the disk images have been downloaded?
> > > > 
> > > > 
> > > > Is this on the right issue? There aren't any disk images or downloading here.
> > > 
> > > 
> > > The `fiwalk_test.cpp` that is now part of the unit test requires disk images.
> > 
> > 
> > Oh. I had not realized that `runner` was not yet being run as part of `make check`. I'll back that part out here and fix that in another PR.
> 
> Thanks. I will likely have `runner` run the to-be-written python program that downloads the disk images if they do not exist.

I advise against that. It would be simpler to have a shell script do the downloads and run the tests.

**Comment by simsong** [comment]
> I advise against that. It would be simpler to have a shell script do the downloads and run the tests.

That is cleaner! We will do that.

**Comment by uckelman-sf** [comment]
@simsong Ready for review.

File: Makefile.am [hunk 1]
```diff
@@ -166,7 +168,7 @@ noinst_LTLIBRARIES = \
 	tsk/vs/libtskvs.la
 
 tsk_auto_libtskauto_la_CFLAGS = $(AM_CFLAGS) -Wmultichar -Wsign-promo -Wno-overloaded-virtual
-tsk_5uto_libtskauto_la_CXXFLAGS = $(AM_CXXFLAGS) -Wextra -Wmultichar -Wsign-promo -Wno-overloaded-virtual
+tsk_auto_libtskauto_la_CXXFLAGS = $(AM_CXXFLAGS) -Wextra -Wmultichar -Wsign-promo -Wno-overloaded-virtual
 tsk_auto_libtskauto_la_SOURCES = \
 	tsk/auto/auto.cpp \
 	tsk/auto/auto_db.cpp \
```
Comment (simsong) [inline Makefile.am:174]: Remember that we are not testing `tsk_auto` in any way...

Comment (uckelman-sf) [reply Makefile.am:174]: The img_dump tests are testing tsk_auto.

This also doesn't change the flags with which libtskauto is compiled.

**Pull request merged**